File: apps/server/src/main/java/com/gainzos/server/entities/WorkoutItem.java
package com.gainzos.server.entities;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "workout_items")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Builder
public class WorkoutItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @Column(name = "exercise_id", nullable = false)
    private Exercise exercise;

    @Column(name = "sets", nullable = false)
    private Integer sets;

    @Column(name = "reps", nullable = false)
    private Integer reps;

    @Column(name = "duration_seconds", nullable = false)
    private Integer durationSeconds;

    @Column(name = "rest_time_seconds", nullable = false)
    private Integer restTimeSeconds;

    @Column(name = "weight", nullable = true)
    private Double weight;
}

File: apps/native/lib/types/workout-item.ts

import { ExerciseSchema } from "./exercise";
import { z } from "zod";

export const WorkoutItemSchema = z.object({
    id: z.number(),
    exercise: ExerciseSchema,
    sets: z.number().int().positive(),
    reps: z.number().int().positive(),
    durationSeconds: z.number().int().positive(),
    restTimeSeconds: z.number().int().positive(),
    weight: z.number().positive(),
});

export type WorkoutItem = z.infer<typeof WorkoutItemSchema>;